Neos.Neos/Classes/Domain/Service/WorkspaceService.php

```diff
@@ -0,0 +1,365 @@
+            WorkspaceDescription::empty(),
+            WorkspaceName::forLive(),
+            $user->getId(),
+        );
+    }
+
+    /**
+     * Determines the permission the given user has for the specified workspace {@see WorkspacePermissions}
+     */
+    public function getWorkspacePermissionsForUser(ContentRepositoryId $contentRepositoryId, WorkspaceName $workspaceName, User $user): WorkspacePermissions
+    {
+        $userIsAdministrator = $this->userService->currentUserIsAdministrator();
+        $workspaceMetadata = $this->loadWorkspaceMetadata($contentRepositoryId, $workspaceName);
+        if ($workspaceMetadata === null) {
+            return WorkspacePermissions::create(false, false, $userIsAdministrator);
+        }
+        if ($workspaceMetadata->ownerUserId !== null && $workspaceMetadata->ownerUserId->equals($user->getId())) {
+            return WorkspacePermissions::all();
+        }
+
+        $userWorkspaceRole = $this->loadWorkspaceRoleOfUser($contentRepositoryId, $workspaceName, $user);
+        if ($userWorkspaceRole === null) {
+            return WorkspacePermissions::create(false, false, $userIsAdministrator);
+        }
+        return WorkspacePermissions::create(
+            read: $userWorkspaceRole->isAtLeast(WorkspaceRole::COLLABORATOR),
+            write: $userWorkspaceRole->isAtLeast(WorkspaceRole::COLLABORATOR),
+            manage: $userIsAdministrator || $userWorkspaceRole->isAtLeast(WorkspaceRole::MANAGER),
+        );
+    }
+
+    // ------------------
+
+    private function loadWorkspaceMetadata(ContentRepositoryId $contentRepositoryId, WorkspaceName $workspaceName): ?WorkspaceMetadata
+    {
+        $table = self::TABLE_NAME_WORKSPACE_METADATA;
+        $query = <<<SQL
+            SELECT
+                *
+            FROM
+                {$table}
+            WHERE
+                content_repository_id = :contentRepositoryId
+                AND workspace_name = :workspaceName
+        SQL;
+        try {
+            $metadataRow = $this->dbal->fetchAssociative($query, [
+                'contentRepositoryId' => $contentRepositoryId->value,
+                'workspaceName' => $workspaceName->value,
+            ]);
+        } catch (DbalException $e) {
+            throw new \RuntimeException(sprintf(
+                'Failed to fetch metadata for workspace "%s" (Content Repository "%s), please ensure the database schema is up to date. %s',
+                $workspaceName->value,
+                $contentRepositoryId->value,
+                $e->getMessage()
+            ), 1727782164, $e);
+        }
+        if (!is_array($metadataRow)) {
+            return null;
+        }
+        assert(is_string($metadataRow['title']));
+        assert(is_string($metadataRow['description']));
+        assert(is_string($metadataRow['classification']));
+        assert(is_null($metadataRow['owner_user_id']) || is_string($metadataRow['owner_user_id']));
```

Comment:
i dont get the sense behind this ... php would just throw an exception either way? And that is likely more descriptive than the error here. I would just rather not have this logic here duplicated for the case it being faulty.

Reply:
without those, PHPStand will complain

Reply:
no it won't :)
I'll remove it